content/en/llm_observability/setup/sdk/nodejs.md

```diff
@@ -0,0 +1,713 @@
+{{< /code-block >}}
+
+### Task span
+
+To trace an LLM span, specify the span kind as `task`, with optionally specifying the following arguments.
+
+#### Arguments
+
+`name`
+: optional - _string_
+<br/>The name of the operation. If not provided, `name` defaults to the name of the traced function.
+
+`sessionId`
+: optional - _string_
+<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.
+
+`mlApp`
+: optional - _string_
+<br/>The name of the ML application that the operation belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information.
+
+#### Example
+
+{{< code-block lang="javascript" >}}
+function sanitizeInput () {
+  ... // user application logic
+  return
+}
+sanitizeInput = llmobs.wrap('task', sanitizeInput)
+{{< /code-block >}}
+
+### Embedding span
+
+To trace an LLM span, specify the span kind as `embedding`, with optionally specifying the following arguments.
+
+**Note**: Annotating an embedding span's input requires different formatting than other span types. See [Annotating a span](#annotating-a-span) for more details on how to specify embedding inputs.
+
+#### Arguments
+
+`modelName`
+: required - _string_
+<br/>The name of the invoked LLM.
+
+`name`
+: optional - _string_
+<br/>The name of the operation. If not provided, `name` is set to the name of the traced function.
+
+`modelProvider`
+: optional - _string_ - **default**: `"custom"`
+
+`sessionId`
+: optional - _string_
+<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.
+
+`mlApp`
+: optional - _string_
+<br/>The name of the ML application that the operation belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information.
+
+#### Example
+
+{{< code-block lang="javascript" >}}
+function performEmbedding () {
+  ... // user application logic
+  return
+}
+performEmbedding = llmobs.wrap('embedding', { modelName: 'text-embedding-3', modelProvider: 'openai' }, performEmbedding)
+{{< /code-block >}}
+
+### Retrieval span
+
+To trace an LLM span, specify the span kind as `retrieval`, with optionally specifying the following arguments.
+
+**Note**: Annotating a retrieval span's output requires different formatting than other span types. See [Annotating a span](#annotating-a-span) for more details on how to specify retrieval outputs.
+
+#### Arguments
+
+`name`
+: optional - _string_
+<br/>The name of the operation. If not provided, `name` defaults to the name of the traced function.
+
+`sessionId`
+: optional - _string_
+<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.
+
+`mlApp`
+: optional - _string_
+<br/>The name of the ML application that the operation belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information.
+
+#### Example
```

Comment:
Small nit (realized it's the same issue with the python page): can we add 1 line here explaining this also includes the annotation example? (and same for the python page)